src/server/controlchan/commands/pasv.rs

```diff
@@ -119,10 +125,40 @@ where
             }
         });
 
-        //Ok(Reply::None)
         Ok(Reply::new_with_string(
             ReplyCode::EnteringPassiveMode,
             format!("Entering Passive Mode ({},{},{},{},{},{})", octets[0], octets[1], octets[2], octets[3], p1, p2),
         ))
     }
+
+    // For proxy mode we prepare the session and let the proxy loop know (via channel) that it
+    // should choose a data port and check for connections on it.
+    async fn handle_proxy_mode<S, U>(&self, args: CommandContext<S, U>, mut tx: ProxyLoopSender<S, U>) -> Result<Reply, ControlChanError>
+    where
+        U: UserDetail + 'static,
+        S: 'static + storage::StorageBackend<U> + Sync + Send,
+        S::File: tokio::io::AsyncRead + Send,
+        S::Metadata: storage::Metadata,
+    {
+        tx.send(ProxyLoopMsg::AssignDataPortCommand(args.session.clone())).await.unwrap();
```

Comment:
A ok. 

b) was not needed since this is happening on the control loop task so only when this returns will the client be let know about it.. but i swapped it anyways